manticore/core/manticore.py

```diff
@@ -357,6 +357,53 @@ def __init__(self, initial_state, workspace_url=None, outputspace_url=None, **kw
         # Workers will use manticore __dict__ So lets spawn them last
         self._workers = [self._worker_type(id=i, manticore=self) for i in range(consts.procs)]
         self._is_main = True
```

Comment:
Is the is_main logic correct?
`
take_snapshot should only work from main. If is_main works maybe we should use that instead of just `at_not_running`

Reply:
It WAS not working. Fixed it and added tests for it. (is_main)